zokrates_core/src/typed_absy/mod.rs

```diff
@@ -1051,11 +1051,27 @@ impl<'ast, T, E> ConditionalExpression<'ast, T, E> {
 impl<'ast, T: fmt::Display, E: fmt::Display> fmt::Display for ConditionalExpression<'ast, T, E> {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self.kind {
-            ConditionalKind::IfElse => write!(
-                f,
-                "if {} {{ {} }} else {{ {} }}",
-                self.condition, self.consequence, self.alternative
-            ),
+            ConditionalKind::IfElse => {
+                let consequence = self.consequence.to_string();
+                let alternative = self.alternative.to_string();
+                let is_block = consequence.starts_with("{") && alternative.starts_with("{");
```

Comment:
This looks hacky, maybe there is a better way? Before `to_string` we have enums which seem cleaner for this sort of thing

Reply:
I looked into it and the issue doesn't seem to be with the formatting code, but rather with the fact that the branch isolator currently wraps blocks in blocks. It should keep blocks as is, and wrap everything else in a block. Then formatting can be done the natural way.